Runtime/Components/CinemachineSplineDolly.cs

```diff
@@ -0,0 +1,399 @@
+                m_PathPosition += m_AutoDolly.m_PositionOffset;
+            }
+            float newPathPosition = m_PathPosition;
+
+            if (deltaTime >= 0 && VirtualCamera.PreviousStateIsValid)
+            {
+                float maxUnit = 1; // we are always using normalized unit [0-1]
+                {
+                    float prev = m_PreviousNormalizedPathPosition;
+                    float next = newPathPosition;
+                    if (pathSpline.Closed && Mathf.Abs(next - prev) > maxUnit / 2)
+                    {
+                        if (next > prev)
+                            prev += maxUnit;
+                        else
+                            prev -= maxUnit;
+                    }
+                    m_PreviousNormalizedPathPosition = prev;
+                    newPathPosition = next;
+                }
+
+                // Apply damping along the path direction
+                float offset = m_PreviousNormalizedPathPosition - newPathPosition;
+                offset = Damper.Damp(offset, m_Damping.z, deltaTime);
+                newPathPosition = m_PreviousNormalizedPathPosition - offset;
+            }
+            m_PreviousNormalizedPathPosition = newPathPosition;
+            m_Track.Evaluate(newPathPosition, 
+                out var localPosition, out var localTangent, out var localUp);
+            Vector3 newCameraPos = localPosition;
+            var newPathOrientation = 
+                Vector3.SqrMagnitude(localTangent) == 0 || Vector3.SqrMagnitude(localUp) == 0 ? 
+                    Quaternion.identity : Quaternion.LookRotation(localTangent, localUp);
+            
+            if (RollOverrideExtension != null && RollOverrideExtension.enabled)
+            {
+                float roll = RollOverrideExtension.RollOverride.Evaluate(pathSpline, newPathPosition, 
+                    PathIndexUnit.Normalized, new UnityEngine.Splines.Interpolators.LerpFloat());
+                var rollRotation = Quaternion.AngleAxis(-roll, localTangent);
+                newPathOrientation = Quaternion.LookRotation(localTangent, rollRotation * localUp);
+            }
+
+            // Apply the offset to get the new camera position
+            var offsetX = newPathOrientation * Vector3.right;
+            var offsetY = newPathOrientation * Vector3.up;
+            var offsetZ = newPathOrientation * Vector3.forward;
+            newCameraPos += m_PathOffset.x * offsetX;
+            newCameraPos += m_PathOffset.y * offsetY;
+            newCameraPos += m_PathOffset.z * offsetZ;
+
+            // Apply damping to the remaining directions
+            if (deltaTime >= 0 && VirtualCamera.PreviousStateIsValid)
+            {
+                Vector3 currentCameraPos = m_PreviousCameraPosition;
+                Vector3 delta = (currentCameraPos - newCameraPos);
+                Vector3 delta1 = Vector3.Dot(delta, offsetY) * offsetY;
+                Vector3 delta0 = delta - delta1;
```

Comment:
are you erroneously combining deltaX and deltaZ here?  You already damped deltaZ, I think

Reply:
I'll look into in depth. 
I used the code from TrackedDolly. I assumed it was correct.

Reply:
Yes, that's the case. deltaX and deltaZ are damping the same thing.

Reply:
But, z Damping only damps on the path, because we use the Z damped value to get the position on the spline. Then we apply damping X Y to this value. So I think this is correct.